Changes for 2010-10-03:
 * Have fonts that contain no truetype instructions use autohint by default

diff -ru freetype-2.4.2.orig/src/base/ftobjs.c freetype-2.4.2/src/base/ftobjs.c
--- freetype-2.4.2.orig/src/base/ftobjs.c 2010-08-06 13:02:15.000000000 -0500
+++ freetype-2.4.2/src/base/ftobjs.c  2010-10-02 21:45:21.349278594 -0500
@@ -562,6 +562,9 @@
     FT_Bool       autohint = FALSE;
     FT_Module     hinter;

+TT_Face face2=(TT_Face)face;
+
+

     if ( !face || !face->size || !face->glyph )
       return FT_Err_Invalid_Face_Handle;
@@ -627,8 +630,15 @@


         if ( mode == FT_RENDER_MODE_LIGHT             ||
-             face->internal->ignore_unpatented_hinter )
+             face->internal->ignore_unpatented_hinter ||
+             ( face2->max_profile.maxSizeOfInstructions  == 0 ) )
+        {
           autohint = TRUE;
+          /* attempt to force slight hinting here, but doesn't work */
+          /*load_flags &= ~FT_RENDER_MODE_NORMAL;
+          load_flags &= ~FT_LOAD_NO_AUTOHINT;
+          load_flags |= FT_RENDER_MODE_LIGHT;*/
+        }
       }
     }

